superset/db_engine_specs/base.py

```diff
@@ -911,13 +912,18 @@ def column_datatype_to_string(
     ) -> str:
         """
         Convert sqlalchemy column type to string representation.
-        Can be overridden to remove unnecessary details, especially
-        collation info (see mysql, mssql).
+        By default removes collation and character encoding info to avoid unnecessarily
+        long datatypes.
 
         :param sqla_column_type: SqlAlchemy column type
         :param dialect: Sqlalchemy dialect
         :return: Compiled column type
         """
+        sqla_column_type = sqla_column_type.copy()
```

Comment:
I've never seen this pattern of copying a variable and renaming it to itself, but I guess it works for this purpose.

Reply:
Slightly dirty, I agree. I just didn't want to mutate the original object, and also didn't see the need to hang on to the original reference, hence this questionable solution 😁